### PR TITLE
Ice Ball and Rollout do not double BP when hitting multiple times/turn

### PR DESCRIPTION
**Pull request by ascriptmaster**
Instead, it now doubles BP after each complete execution of the move.

### CODE REVIEW

**Comment by ascriptmaster** [comment]
Still waiting on @Marty-D for this

**Comment by Marty-D** [comment]
I just realized now that this doesn't actually address the real issue, which is that Parental Bond does not affect Ice Ball or Rollout. They can't become multi-hit moves at all. Sorry for the confusion, I thought you were around for that part of the conversation re: Parental Bond.

Something like
```javascript
onTry: function (source, target, move) {
	delete move.multihit;
},
```
should do the trick.

**Comment by ascriptmaster** [comment]
So behavior should be like this?

I'd update the PR title but pushing my changes was the last thing I could do on my computer before it ran out of battery, and I can't figure out how on mobile.

**Comment by Marty-D** [comment]
Yeah, that's the right behaviour, but I'm not sure if the Ice Ball/Rollout multi-hit stuff is worth implementing. Are we supposed to be covering hypothetical situations that involve actual hacking of move mechanics? That seems like a ridiculously tall order. @Zarel can weigh in on this.

File: data/moves.js [hunk 1]
```diff
@@ -11324,15 +11324,15 @@ exports.BattleMovedex = {
 			onLockMove: 'rollout',
 			onStart: function () {
 				this.effectData.hitCount = 1;
+				this.effectData.trueDuration = 5;
```
Comment (Zarel) [inline data/moves.js]: Don't call it `trueDuration` if it's not a turn count. Call it `hits` or whatever else it's actually counting.

**Comment by Zarel** [comment]
I don't see a need to cover situations that involve hacking move mechanics.

**Comment by Marty-D** [comment]
Come to think of it, hacking move mechanics (even just a move's type) doesn't affect Wi-Fi play at all, so this kind of thing is impossible to do against another player anyway. Please remove.

**Comment by Zarel** [comment]
I mean, I'm fine with matching the in-game mechanic in hacked situations if it leads to simpler code that matches the real code more closely, but if it's just a hack, don't bother.

File: data/moves.js (hunk 1 above)
Comment (ascriptmaster) [inline data/moves.js]: It is a turn count though, lol

**Comment by ascriptmaster** [comment]
Updated to not care about hacked mechanics. I'm a pedant though, so if Rollout hits more than 5 hits without resetting somehow, the other hits are 30 BP now.

**Comment by Marty-D** [comment]
Good call, thanks! :+1:

File: data/moves.js (hunk 1 above)
Comment (Zarel) [inline data/moves.js]: Wait, if it is a turn count, why is it different from `duration`?

Comment (Zarel) [inline data/moves.js]: Maybe use a different word than "true" in front of "duration"? The way the variable is currently named, I don't have a good idea of what exactly it tracks.

**Pull request merged**